File: 박인우/[프로그래머스] 크레인 인형뽑기.py
def solution(board, moves):
    answer = 0
    stack = []
    for x in moves:
        x = x-1
        for y in range(len(board)):
            now = board[y][x]
            if now > 0:
                board[y][x] = 0
                if len(stack) and stack[-1] == now:
                    stack.pop()
                    answer += 2
                else:
                    stack.append(now)
                break
            else:
                continue
    return answer

File: 박인우/0207-0213/[프로그래머스] 크레인 인형뽑기.swift
import Foundation

func solution(_ board:[[Int]], _ moves:[Int]) -> Int {
    var board = board
    var stack: [Int] = []
    var count = 0
    for move in moves {
        var x = move - 1
        var y = 0
        while y < board.count && board[y][x] == 0 {
            y += 1
        }
        if y < board.count {
            var kind = board[y][x]
            board[y][x] = 0
            if stack.count > 0 && stack.last! == kind {
                stack.removeLast()
                count += 2
            } else {
                stack.append(kind)
            }
        } 
    }
    return count
}
